modules/order/src/OrderStorage.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace Drupal\commerce_order;
+
+use Drupal\commerce\CommerceContentEntityStorage;
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\Core\Cache\CacheBackendInterface;
+use Drupal\Core\Database\Connection;
+use Drupal\Core\Entity\EntityManagerInterface;
+use Drupal\Core\Language\LanguageManagerInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+
+/**
+ * Defines the order storage.
+ */
+class OrderStorage extends CommerceContentEntityStorage {
+
+  /**
+   * The order refresh process.
+   *
+   * @var \Drupal\commerce_order\OrderRefreshInterface
+   */
+  protected $orderRefresh;
+
+  /**
+   * Constructs a new CommerceContentEntityStorage object.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeInterface $entity_type
+   *   The entity type definition.
+   * @param \Drupal\Core\Database\Connection $database
+   *   The database connection to be used.
+   * @param \Drupal\Core\Entity\EntityManagerInterface $entity_manager
+   *   The entity manager.
+   * @param \Drupal\Core\Cache\CacheBackendInterface $cache
+   *   The cache backend to be used.
+   * @param \Drupal\Core\Language\LanguageManagerInterface $language_manager
+   *   The language manager.
+   * @param \Symfony\Component\EventDispatcher\EventDispatcherInterface $event_dispatcher
+   *   The event dispatcher.
+   * @param \Drupal\commerce_order\OrderRefreshInterface $order_refresh
+   *   The order refresh process.
+   */
+  public function __construct(EntityTypeInterface $entity_type, Connection $database, EntityManagerInterface $entity_manager, CacheBackendInterface $cache, LanguageManagerInterface $language_manager, EventDispatcherInterface $event_dispatcher, OrderRefreshInterface $order_refresh) {
+    parent::__construct($entity_type, $database, $entity_manager, $cache, $language_manager, $event_dispatcher);
+    $this->orderRefresh = $order_refresh;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function createInstance(ContainerInterface $container, EntityTypeInterface $entity_type) {
+    return new static(
+      $entity_type,
+      $container->get('database'),
+      $container->get('entity.manager'),
+      $container->get('cache.entity'),
+      $container->get('language_manager'),
+      $container->get('event_dispatcher'),
+      $container->get('commerce_order.order_refresh')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function postLoad(array &$entities) {
+    foreach ($entities as $entity) {
```

Comment:
Wondering if it would have made more sense to do this in hook_commerce_order_load(). Each overriden storage is something we'll need to undo at some point, to allow the workflow modules to do their own swapping.